src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.Generation.Types;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace AutoRest.CSharp.Output.Models.Shared
+{
+    internal static class KnownParameters
+    {
+        private static readonly CSharpType RequestContentType = new(typeof(RequestContent));
+        private static readonly CSharpType RequestContentNullableType = new(typeof(RequestContent), true);
+        private static readonly CSharpType RequestContextNullableType = new(typeof(RequestContext), true);
+
+        public static readonly Parameter ClientDiagnostics = new("clientDiagnostics", "The ClientDiagnostics instance to use", new CSharpType(typeof(ClientDiagnostics)), null, true);
+        public static readonly Parameter Pipeline = new("pipeline", "The pipeline instance to use", new CSharpType(typeof(HttpPipeline)), null, true);
+        public static readonly Parameter KeyAuth = new("keyCredential", "The key credential to copy", new CSharpType(typeof(AzureKeyCredential)), null, false);
+        public static readonly Parameter TokenAuth = new("tokenCredential", "The token credential to copy", new CSharpType(typeof(TokenCredential)), null, false);
+
+        public static readonly Parameter RequestContent = new("content", "The content to send as the body of the request.", RequestContentType, null, true);
+        public static readonly Parameter RequestContentNullable = new("content", "The content to send as the body of the request.", RequestContentNullableType, /*Constant.Default(RequestContentNullableType)*/null, false);
+
+        public static readonly Parameter RequestContext = new("context", "The request context.", RequestContextNullableType, Constant.Default(RequestContextNullableType), false);
```

Comment:
@annelo-msft , we probably need better description for `RequestContext` (and maybe you want better descriptions for other parameters as well).

Reply:
What about "The request context, which can override default behaviors on the request on a per-call basis." ?